### PR TITLE
github/contributing: updates to onboarding docs

### PR DESCRIPTION
**Pull request by krantzinator**
This PR fills in some info that will help folks onboard as contributors.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -79,13 +79,26 @@ that must be on your PATH to build Waypoint. This
 [repository version](https://github.com/kevinburke/go-bindata/) may be installed with:
 `go get -u github.com/kevinburke/go-bindata/...`
 
-Next, clone this repository and then run `make bin`. In a few moments, you'll have a working
-`waypoint` executable in the Waypoint repository root and `$GOPATH/bin`:
+Next, clone this repository and then run the following commands:
+* `make bin` will build the binary for your local machine's os/architecture
+* (optional) `make install` will move that executable binary to `$GOPATH/bin/waypoint`
```
Comment (catsby) [inline .github/CONTRIBUTING.md:84]:
```suggestion
* (optional) `make install` will copy that executable binary to `$GOPATH/bin/waypoint`
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -79,13 +79,26 @@ that must be on your PATH to build Waypoint. This
...
+* `make docker/server` will build the server with the tag `waypoint:dev`
```
Comment (catsby) [inline .github/CONTRIBUTING.md:85]:
```suggestion
* `make docker/server` will build the docker image of the server with the tag `waypoint:dev`
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -79,13 +79,26 @@ that must be on your PATH to build Waypoint. This
...
 
->Note: `make bin` will build for your local machine's os/architecture.
+Once those steps are complete, you can install the waypoint server you just built. To do
+this on docker you would run:
+```
+waypoint install -platform=docker -accept-tos -docker-server-image=waypoint:dev
+```
+
+>Note: If you don't run `make install` then you should use `path/to/waypoint`
```
Comment (catsby) [inline .github/CONTRIBUTING.md:93]:
```suggestion
>Note: If you didn't run `make install` then you should use `path/to/waypoint` 
```

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -106,6 +119,24 @@ If the scope of the code change requires it, follow the [Changelog Guide](/.gith
...
+1. Title the PR with a helpful prefix following the pattern `parent/child`, 
+e.g. `builtin/k8s` or `internal/core`
+1. Include helpful information in the description
+   * For a bug fix, explain or show an example of the behavior before and 
+  after the change
+   * If applicable, include information on how to test manually
+1. Assign the `backport/[version].x` label for PRs that should be included 
+in an upcoming .x release (e.g. `backport/0.3.x`)
+1. Assign the `backport/website` label for PRs that should be published 
+immediately to the website
```
Comment (catsby) [inline .github/CONTRIBUTING.md:133]: Nit/question: Assuming the audience of this contributing doc is both public OSS contributors as well as developers at HashiCorp, I feel like maybe we should omit the `backport` labeling parts here, and leave those as a internal task HashiCorp devs are responsible for. What do you think?

**Review by catsby — APPROVED** [review]
Minor nits and a question about labels, otherwise LGTM 👍

**Review by briancain — APPROVED** [review]
👍🏻 to @catsby 's comments, but looks great!

Doesn't have to be in this PR, but it would be good to include a section on `make gen/server` for generating protobufs on the server side.

**Pull request merged**